src/prefect/task_engine.py

```diff
@@ -115,7 +116,7 @@ class BaseTaskRunEngine(Generic[P, R]):
     # holds the return value from the user code
     _return_value: Union[R, Type[NotSet]] = NotSet
     # holds the exception raised by the user code, if any
-    _raised: Union[Exception, Type[NotSet]] = NotSet
+    _raised: Union[BaseException, Type[NotSet]] = NotSet
```

Comment:
I think we still want to keep `Exception` here.
```suggestion
    _raised: Union[Exception, BaseException, Type[NotSet]] = NotSet
```